tests/rest_api/test_configurations.py

```diff
@@ -653,26 +656,220 @@ def test_resource_hints():
         ],
     }
 
+    with patch.object(dlt, "resource", wraps=dlt.resource) as mock_resource_constructor:
+        rest_api_resources(config)
+        mock_resource_constructor.assert_called_once()
+        expected_kwargs = {
+            "table_name": "a_table",
+            "max_table_nesting": 2,
+            "write_disposition": "merge",
+            "columns": {"a_text": {"name": "a_text", "data_type": "text"}},
+            "primary_key": "a_pk",
+            "merge_key": "a_merge_key",
+            "schema_contract": {"tables": "evolve"},
+            "table_format": "iceberg",
+            "selected": False,
+        }
+        for arg in expected_kwargs.items():
+            _, kwargs = mock_resource_constructor.call_args_list[0]
+            assert arg in kwargs.items()
+
+
+def test_two_resources_dependent_on_one_parent_resource():
+    config: RESTAPIConfig = {
+        "client": {
+            "base_url": "https://api.example.com",
+        },
+        "resources": [
+            "users",
+            {
+                "name": "user_details",
+                "endpoint": {
+                    "path": "user/{user_id}/",
+                    "params": {
+                        "user_id": {
+                            "type": "resolve",
+                            "field": "id",
+                            "resource": "users",
+                        },
+                    },
+                },
+            },
+            {
+                "name": "meetings",
+                "endpoint": {
+                    "path": "meetings/{user_id}/",
+                    "params": {
+                        "user_id": {
+                            "type": "resolve",
+                            "field": "id",
+                            "resource": "users",
+                        },
+                    },
+                },
+            },
+        ],
+    }
+    resources = rest_api_resources(config)
+
+    resource_names = [resource.name for resource in resources]
+    assert ["meetings", "user_details", "users"] == sorted(resource_names)
```

Comment:
Let's also check if the dependency has been established by checking `resource._pipe.parent.name`